apiserver/facades/agent/uniter/newlxdprofile.go

```diff
@@ -211,12 +217,23 @@ func (u *LXDProfileAPIv2) LXDProfileName(args params.Entities) (params.StringRes
 			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
 			continue
 		}
-		unit, machine, err := u.getLXDProfileUnitMachineV2(tag)
+		unit, _, err := u.getLXDProfileUnitMachineV2(tag)
 		if err != nil {
 			result.Results[i].Error = apiservererrors.ServerError(err)
 			continue
 		}
-		name, err := u.getOneLXDProfileName(unit, machine)
+
+		machineTagID, err := unit.AssignedMachineId()
+		if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+		machineUUID, err := u.machineService.GetMachineUUID(ctx, coremachine.Name(machineTagID))
+		if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+		name, err := u.getOneLXDProfileName(ctx, unit, machineUUID)
```

Comment:
The name of this function is a bit weird, why is it get _one_ LXD profile name. It seems to get the LXD profile for the application, can an application only have one? Or is it just getting a random one of the applicaitons.

I know this function name isn't part of the PR but it would be good to clarify what the intention of this function is, either by changing the name or adding a comment.